QAChat/Data_Processing/data_preprocessor.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Jesse Tim Palarus
+# SPDX-FileCopyrightText: 2023 Amela Pucic
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import List
+
+from document_embedder import DataInformation
+
+
+class DataPreprocessor(ABC):
+    @abstractmethod
+    def load_preprocessed_data(self, before: datetime, after: datetime) -> List[DataInformation]:
+        """
+        Loads preprocessed data of a specific type that was created or modified
```

Comment:
The timeframe is of when the data were added to the database or is it based of information like the last changed attribute.
Is there a way to diable this, as I dont see a possible usage for timeframes in our usage scenario.

Reply:
The time frame refers to when an element was created or last edited.

The main reason for this is that if, for example, only a few entries are changed on the confluence wiki or only a few are added, then of course only the changes should be recreated in the database. Therefore this method only asks for the changes.

You can basically disable it by setting before to a time in the future and after to something like 1.1.1970, but then the complete data is recreated in the database, which will hurt performance.

Reply:
I see makes sense. Didnt think about that.